ksqldb-engine/src/main/java/io/confluent/ksql/schema/ksql/inference/DefaultSchemaInjector.java

```diff
@@ -121,6 +122,15 @@ private SchemaAndId getValueSchema(
     return result.schemaAndId.get();
   }
 
+  private static boolean valueFormatDoesNotSupportSchemaInference(
+      final ConfiguredStatement<CreateSource> statement
+  ) {
+    final Format valueFormat = FormatFactory
+        .of(statement.getStatement().getProperties().getFormatInfo());
```

Comment:
This might not be present at this step. Should this also delegate to TopicFactory? Feels a bit odd since there's no topic being created.

Reply:
Also, we still need to throw on unsupported key formats somewhere. For C*AS statements we could do it in the Analyzer, but I wonder where it makes sense to do it for C* statements. Should we do that here?

In https://github.com/confluentinc/ksql/pull/6194 this is happening for both C* and C*AS statements in EngineExecutor#plan(). If we want to do it there we'll need to add logic for finding the default formats there as well, which doesn't feel right.

Reply:
Another argument in favor of the injector is that the statement shown to the user currently has everything else filled in explicitly: schemas (in the case of schema inference), resolved types (in the case of custom types), and number of topic replicas/partitions. The statement reflects exactly what the engine executed. Not populating default key/value formats would break pattern here.

In terms of arguments against the injector:
- one less injector, definitely a plus
- ??

Both approaches involve a decent amount of headache around the feature flag. In this PR we'll have to add checks somewhere to validate that default value formats (and probably explicit key formats also) can't be used unless the feature flag is enabled. Looks like we'll have to do that separately for `C*` and `C*AS` statements under this design.